packages/edit-site/src/components/global-styles/global-styles-provider.js

```diff
@@ -45,22 +45,35 @@ const cleanEmptyObject = ( object ) => {
 };
 
 function useGlobalStylesUserConfig() {
-	const { globalStylesId, settings, styles } = useSelect( ( select ) => {
-		const _globalStylesId =
-			select( coreStore ).__experimentalGetCurrentGlobalStylesId();
-		const record = _globalStylesId
-			? select( coreStore ).getEditedEntityRecord(
-					'root',
-					'globalStyles',
-					_globalStylesId
-			  )
-			: undefined;
-		return {
-			globalStylesId: _globalStylesId,
-			settings: record?.settings,
-			styles: record?.styles,
-		};
-	}, [] );
+	const { globalStylesId, hasGlobalStyles, settings, styles } = useSelect(
+		( select ) => {
+			const { getEditedEntityRecord, getEntityRecords } =
+				select( coreStore );
+			const _globalStylesId =
+				select( coreStore ).__experimentalGetCurrentGlobalStylesId();
+			const record = _globalStylesId
+				? getEditedEntityRecord(
+						'root',
+						'globalStyles',
+						_globalStylesId
+				  )
+				: undefined;
+			const activeThemes = getEntityRecords( 'root', 'theme', {
+				status: 'active',
+			} );
```

Comment:
The endpoint in the prefetched and the same selector is resolved in `__experimentalGetCurrentGlobalStylesId`, so this shouldn't cause any delays.